src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/overlays/mediacropdetails.controller.js

```diff
@@ -56,4 +57,8 @@
                 }
             }
 
+            function cropSet() {
+                var model = $scope.model;
+                return (model.cropSize || {}).width && model.target.thumbnail;
```

Comment:
@lars-erik I didn't look very closely when merging this. This would return either an empty string or the `model.target.thumbnail` value. I've changed it to check for a configure width/height:  https://github.com/umbraco/Umbraco-CMS/commit/e1ef5f4e6b6b28715d965a9a3d2dac951f251de6#diff-c9ab1fa1ff35ffa6b4551dd5b51a95aeR62

I must admit I don't actually understand what the `|| {}` means in `(model.cropSize || {})`. I've left it for now, but can you clarify please?

Reply:
The `(x.y || {}).z` is the same as `x.y && x.y.z` - in cases where y is undefined, the RHS assigns an empty object so the check for z won't fail.

Given that this case is checking truthiness, not using the actual value of z, either way works the same - if we needed the value of z, it's a handy pattern.

Javascript does have optional chaining too, so we could do `x?.y?.z` if the build had the required transpiling for browser support...

Reply:
@nathanwoulfe Thanks! I guess I kinda mixed and matched this in my update then 🙈 

https://github.com/umbraco/Umbraco-CMS/commit/e1ef5f4e6b6b28715d965a9a3d2dac951f251de6#diff-c9ab1fa1ff35ffa6b4551dd5b51a95aeR62

Are you saying that: `(model.cropSize || {}).width !== undefined` will give the same result as `(model.cropSize || {}).width`?

Reply:
@nul800sebastiaan JavaScript works with truthy and falsy things.
Null, undefined, 0, undefined, "" etc. are all "falsy".
So you can just test if it's anything except those.
|| is just the same as C# ??.

Reply:
Should do, yes, thanks to javascript coercing to boolean, but I'd keep the version with the explicit comparison since it's easier to read.